ts/screens/profile/PreferencesScreen.tsx

```diff
@@ -149,154 +103,89 @@ class PreferencesScreen extends React.Component<PreferencesScreenProps> {
-        headerActionsProp={{ showHelp: true }}
-        faqCategories={["profile", "privacy", "authentication_SPID"]}
-      >
-        <FlatList
-          scrollEnabled={false}
-          keyExtractor={(item: PreferencesNavListItem, index: number) =>
-            `${item.value}-${index}`
-          }
-          contentContainerStyle={{
-            paddingHorizontal: IOVisualCostants.appMarginDefault
-          }}
-          data={filteredPreferencesNavListItems}
-          renderItem={renderPreferencesNavItem}
-          ItemSeparatorComponent={() => <Divider />}
+      onPress: checkPermissionThenGoCalendar
+    },
+    {
+      // Service Contacts
+      value: I18n.t("profile.preferences.list.service_contact.title"),
+      description: I18n.t("profile.preferences.list.service_contact.subtitle"),
+      onPress: navigateToServicePreferenceScreen
+    },
+    {
+      // Email forwarding
+      value: I18n.t("profile.preferences.list.send_email_messages.title"),
+      description: I18n.t(
+        "profile.preferences.list.send_email_messages.subtitle"
+      ),
+      onPress: navigateToEmailForwardingPreferenceScreen
+    },
+    {
+      // Language
+      value: I18n.t("profile.preferences.list.language.title"),
+      description: I18n.t("profile.preferences.list.language.subtitle"),
+      onPress: navigateToLanguagePreferenceScreen
+    }
+  ];
+
+  const renderPreferencesNavItem = ({
+    item: { value, description, onPress, testID, condition }
+  }: ListRenderItemInfo<PreferencesNavListItem>) => {
+    // If condition is either true or undefined, render the item
+    if (condition !== false) {
+      return (
+        <ListItemNav
+          accessibilityLabel={`${value} ${description}`}
+          value={value}
+          description={description}
+          onPress={onPress}
+          testID={testID}
         />
-      </RNavScreenWithLargeHeader>
-    );
-  }
-}
-
-function mapStateToProps(state: GlobalState) {
-  return {
-    preferredLanguage: preferredLanguageSelector(state),
-    profileServicePreferenceMode: profileServicePreferencesModeSelector(state),
-    isEmailEnabled: isEmailEnabledSelector(state),
-    isInboxEnabled: isInboxEnabledSelector(state),
-    isCustomEmailChannelEnabled: isCustomEmailChannelEnabledSelector(state),
-    preferredCalendar: state.persistedPreferences.preferredCalendar
+      );
+    } else {
+      return null;
+    }
   };
-}
-
-const mapDispatchToProps = (_: Dispatch) => ({
-  navigateToServiceContactPreferenceScreen: () =>
-    navigateToServicePreferenceScreen(),
-  navigateToEmailForwardingPreferenceScreen: () =>
-    navigateToEmailForwardingPreferenceScreen(),
-  navigateToCalendarPreferenceScreen: () =>
-    navigateToCalendarPreferenceScreen(),
-  navigateToLanguagePreferenceScreen: () => navigateToLanguagePreferenceScreen()
-});
+  const filteredPreferencesNavListItems = preferencesNavListItems.filter(
+    item => item.condition !== false
+  );
```

Comment:
my initial intention was to eliminate this 'condition' but I thought it might be useful in the future. Do you think it will no longer be necessary?

Reply:
Since today it's not necessary I suggest to remove it. If in the future it'll be necessary again we'll reintroduce it. This way we spare LOC (line of codes) and a non necessary filter operation. What do you think?

Reply:
done! ✅